agoradatatools/etl/transform.py

```diff
@@ -137,6 +151,105 @@ def transform_network(datasets: dict):
 
     return merged
 
+
+def transform_gene_metadata(datasets: dict, adjusted_p_value_threshold, protein_level_threshold):
+    '''
+    This function will perform transformations and incrementally create a dataset called gene_metadata.
+    Each dataset will be left_joined onto gene_info.
+    '''
+    gene_info = datasets['syn25953363']
+    igap = datasets['syn12514826']
+    eqtl = datasets['syn12514912']
+    proteomics = datasets['syn18689335']
+    brain_expression_change = datasets['syn11914808']
+    rna_change = datasets['syn14237651']
+
+    # remove duplicate ensembl_gene_ids and select columns
+    gene_info = gene_info.groupby('ensembl_gene_id').apply(lambda x: x.nlargest(1, "_version")).reset_index(drop=True)
+    gene_info = gene_info[['ensembl_gene_id', 'symbol', 'name', 'summary', 'alias']]
+
+
+    gene_metadata = pd.merge(left=gene_info, right=igap, how='left', on='ensembl_gene_id')
+    gene_metadata['igap'] = gene_metadata.apply(lambda row: False if row['hgnc_symbol'] is NaN else True, axis=1)
```

Comment:
Minor: Is it more efficient to apply through the entire dataframe or just loop through the one column.

Can probably be

```
gene_metadata['igap'] = ~row['hgnc_symbol'].isnull()
```

Reply:
### NaN
np.nan == float('nan') is **False** 🤷🏼‍♀️  

### pd.DataFrame.apply()
Oh, this is faster than a loop.  A quick comparison [here](https://towardsdatascience.com/how-to-make-your-pandas-loop-71-803-times-faster-805030df4f06).

### .isnull()
I could use isnull().  I just don't know if there's a performance difference here.